.github/workflows/doc.yml

```diff
@@ -0,0 +1,22 @@
+name: doc
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: cachix/install-nix-action@v18
+      - run: nix build .#doc
+      - if: success()
+        uses: crazy-max/ghaction-github-pages@v3
```

Comment:
I generally prefer to go with the "official" actions instead of third-party (though I'm totally on board with `cachix/install-nix-action`). If interested, here's an example using the official `actions/configure-pages` and `actions/deploy-pages` actions: https://github.com/n8henrie/n8henrie-rs/blob/f02a99040fc85f791a63d50af7bf367e24df81dd/.github/workflows/docs.yml, the resulting GH pages: https://n8henrie.com/n8henrie-rs/n8henrie/ (was using it as a playground repo for GH docs).

Reply:
Good call out. I'd rather use some official one too. I'll check these out.

Reply:
Any idea what is going on here https://github.com/ryantm/agenix/actions/runs/4332366715/jobs/7565206043 ?

> Error: Deployment failed, try again later.